src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs

```diff
@@ -469,32 +530,54 @@ public async Task<List<PlatformSummary>> GetElasticBeanstalkPlatformArns()
                         Operator = "=",
                         Type = "PlatformStatus",
                         Values = { "Ready" }
+                        },
+                        new PlatformFilter
+                        {
+                            Operator = "contains",
+                            Type = "PlatformName",
+                            Values = { platformName }
                     }
                 }
             };
-            var response = await HandleException(async () => await beanstalkClient.ListPlatformVersionsAsync(request));
+
+                var platforms = await HandleException(async () => (await beanstalkClient.Paginators.ListPlatformVersions(request).PlatformSummaryList.ToListAsync()));
+
+                // Filter out old test platforms that only internal accounts would be able to see.
+                platforms = platforms.Where(x => !string.IsNullOrEmpty(x.PlatformBranchName)).ToList();
+
+                return platforms;
+            };
+
+            var allPlatformSummaries = new List<PlatformSummary>();
+            if (platformTypes.Contains(BeanstalkPlatformType.Linux))
+            {
+                allPlatformSummaries.AddRange(await fetchPlatforms(".NET Core"));
+            }
+            else if (platformTypes.Contains(BeanstalkPlatformType.Windows))
+            {
+                var windowsPlatforms = await fetchPlatforms("Windows Server");
+                SortElasticBeanstalkWindowsPlatforms(windowsPlatforms);
```

Comment:
could the linux platform benefit from the same behavior? i.e. adding sorting

Reply:
Linux platform doesn't have the issue of having multiple OSes like the Windows platforms do. It is possible in the future when more versions of Amazon Linux are made available through Beanstalk but it will have a different naming pattern to sort by.